### PR TITLE
fix(auth): recover users signup data when them click on terms

### LINKED ISSUE #72: [auth] recover user signup data when he clicks on terms

### PR DESCRIPTION
**Pull request by murosfc**
closes #72 
It is also working if user leaves signup page and returns.

### CODE REVIEW

File: src/app/authentication/signup-page/signup-page.component.html [hunk 1]
```diff
@@ -52,8 +52,14 @@
...
-    Eu li e concordo com os <a routerLink="/terms-of-use">termos de uso</a> e a
-    <a routerLink="/privacy-policy">política de privacidade</a>.
+    Eu li e concordo com os
+    <a (click)="openTermsOrPolicy('/terms-of-use')" tabindex="4"
+      >termos de uso</a
+    >
+    e a
+    <a (click)="openTermsOrPolicy('/privacy-policy')" tabindex="5"
+      >política de privacidade</a
+    >.
```
Comment (rosostolato) [inline src/app/authentication/signup-page/signup-page.component.html:62]: **issue**: it's better to keep the routerLinks and subscribe to form changes to persist to storage with this code:

  constructor() {
    this.form.valueChanges.pipe(takeUntilDestroyed()).subscribe(() => {
      localStorage.setItem('formData', JSON.stringify(this.form.getRawValue()));
    });
  }

File: src/app/authentication/signup-page/signup-page.component.ts [hunk 2]
```diff
@@ -57,7 +58,7 @@ type FormControlGroup<T> = FormGroup<{
   ],
   providers: [provideNgxMask()],
 })
-export class SignupPageComponent {
+export class SignupPageComponent implements OnInit {
```
Comment (rosostolato) [inline src/app/authentication/signup-page/signup-page.component.ts:61]: **issue**: won't need `OnInit` anymore

File: src/app/authentication/signup-page/signup-page.component.ts [hunk 3]
```diff
@@ -176,7 +177,25 @@ export class SignupPageComponent {
         });
       }
     } finally {
+      const formData = localStorage.getItem('formData');
+      if (formData) {
+        localStorage.removeItem('formData');
+      }
```
Comment (rosostolato) [inline src/app/authentication/signup-page/signup-page.component.ts:183]: **discussion**: got it, you're gonna delete on success. Ok, it's a good idea. But I think you can remove the key right away, don't need to check if it exists.

```suggestion
      localStorage.removeItem('formData');
```

File: src/app/authentication/signup-page/signup-page.component.ts [hunk 4]
```diff
@@ -176,7 +177,25 @@ export class SignupPageComponent {
...
+    }
+    this.router.navigate([route]);
+  }
+
+  ngOnInit() {
+    const formData = localStorage.getItem('formData');
+    if (formData) {
+      this.form.patchValue(JSON.parse(formData));
+    }
+  }
```
Comment (rosostolato) [inline src/app/authentication/signup-page/signup-page.component.ts:200]: **issue**: let's replace it with constructor code and change "formData" to "signupForm".

```suggestion
  constructor() {
    const formData = localStorage.getItem('signupForm');
    if (formData) {
      this.form.patchValue(JSON.parse(formData));
    }

    this.form.valueChanges.pipe(takeUntilDestroyed()).subscribe(() => {
      localStorage.setItem(
        'signupForm',
        JSON.stringify(this.form.getRawValue()),
      );
    });
  }
```

**Review by rosostolato — CHANGES_REQUESTED** [review]

**Comment by murosfc** [comment]
Changes implemented as per requested. Loading data also added to constructor.

**Pull request merged**